[iOS] fast/events/touch/ios/long-press-on-(link|image).html fail after 251828@main

whsieh opened the pull request:
#### 1c032700aac4f419369112427362cf51fe99d5af
<pre>
[iOS] fast/events/touch/ios/long-press-on-(link|image).html fail after 251828@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=243114">https://bugs.webkit.org/show_bug.cgi?id=243114</a>

Reviewed by Aditya Keerthi.

After `251828@main`, the `_contextMenuInteraction` is no longer used to present the context menu
when long pressing images and links; instead, we rely on the text interaction assistant&apos;s context
menu interaction.

As a result, the testing hook `-[WKContentView _contentsOfUserInterfaceItem:]` no longer works,
since it still calls `_contextMenuInteraction` directly; fix this by replacing it with a call to the
`-contextMenuInteraction` helper method.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _contentsOfUserInterfaceItem:]):

Canonical link: <a href="https://commits.webkit.org/252750@main">https://commits.webkit.org/252750@main</a>
</pre>

pxlcoder commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -11526,7 +11526,7 @@ - (NSDictionary *)_contentsOfUserInterfaceItem:(NSString *)userInterfaceItem
 #if HAVE(LINK_PREVIEW)
     if ([userInterfaceItem isEqualToString:@"contextMenu"]) {
         auto itemTitles = adoptNS([NSMutableArray<NSString *> new]);
-        [_contextMenuInteraction updateVisibleMenuWithBlock:[&itemTitles](UIMenu *menu) -> UIMenu * {
```
We should consider renaming this variable, given that `self.contextMenuInteraction` doesn't always return it anymore.

whsieh replied:
Yeah, that makes sense (I also wonder if we can simply remove this variable altogether, if `HAVE(TEXT_INTERACTION_WITH_CONTEXT_MENU_INTERACTION)`). @mwyrzykowski, do you know if there are cases on iOS 16 where we still rely on `_contextMenuInteraction` rather than the text interaction assistant's context menu interaction?

mwyrzykowski replied:
yes it should be fine for iOS 16 to remove it

whsieh replied:
Filed https://webkit.org/b/243117 for this — since there isn't an urgent need to do this right away, let's wait until we drop support for iOS 15 in EWS/the OpenSource WebKit build.

whsieh committed:
[iOS] fast/events/touch/ios/long-press-on-(link|image).html fail after 251828@main
https://bugs.webkit.org/show_bug.cgi?id=243114

Reviewed by Aditya Keerthi.

After `251828@main`, the `_contextMenuInteraction` is no longer used to present the context menu
when long pressing images and links; instead, we rely on the text interaction assistant's context
menu interaction.

As a result, the testing hook `-[WKContentView _contentsOfUserInterfaceItem:]` no longer works,
since it still calls `_contextMenuInteraction` directly; fix this by replacing it with a call to the
`-contextMenuInteraction` helper method.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _contentsOfUserInterfaceItem:]):

Canonical link: https://commits.webkit.org/252750@main
